### PR TITLE
Ddb 1492 calculate last update value for measures

### PR DESCRIPTION
**Pull request by mateen-hussain**
PR for Part-1 of story https://technologyprogramme.atlassian.net/browse/DDB-1149
As part of this PR https://technologyprogramme.atlassian.net/browse/DDB-1492:
** we calculate last updated data for measures (grouped and non grouped)
** the column "updatedAt" is set only if a value for measure is added and not when a measure information is changed.

### CODE REVIEW

File: helpers/measures.js [hunk 1]
```diff
@@ -76,6 +76,15 @@ const getMeasureEntitiesFromGroup = (groupEntities, metricId) => {
   return sortedEntities;
 }
 
+const getMaxUpdateAtForMeasures = (measures) => {
+  let maxMeasureUpdatedAt;
+  measures.forEach(measure => {
```
Comment (dtaylor7) [inline helpers/measures.js:81]: a sort without moment would be a faster approach but optional 👍

File: pages/data-entry-entity/measure-edit/MeasureEdit.js [hunk 2]
```diff
@@ -337,7 +333,7 @@ class MeasureEdit extends Page {
     }
 
     const updatedEntites = await this.updateMeasureEntities(formData);
-    return await this.saveMeasureData(updatedEntites, URLHash, { ignoreParents: true, updatedAt: true });
+    return await this.saveMeasureData(updatedEntites, URLHash, { ignoreParents: true, updatedAt: false });
```
Comment (dtaylor7) [inline pages/data-entry-entity/measure-edit/MeasureEdit.js:336]: i believe @leighjohnson-friday changed this original, @leighjohnson-friday can you check this is ok?

File: helpers/measures.js (hunk 1 above)
Comment (mateen-hussain) [reply helpers/measures.js:81]: sort takes O(nlogn) to order. While iterating once through the array is O(n). But definetly an good point to keep in mind 👍

File: helpers/measures.js [hunk 3]
```diff
@@ -76,6 +76,15 @@ const getMeasureEntitiesFromGroup = (groupEntities, metricId) => {
...
+    const entityUpdatedAt = (measure.updatedAt) ? moment(measure.updatedAt) : moment(measure.createdAt);
+    maxMeasureUpdatedAt = (maxMeasureUpdatedAt && maxMeasureUpdatedAt.isSameOrAfter(entityUpdatedAt)) ? maxMeasureUpdatedAt : entityUpdatedAt;
+  })
+  return maxMeasureUpdatedAt
```
[the remark this answers is not in the document]
Comment (mateen-hussain) [reply helpers/measures.js:85]: crap. I miss eslint for stuff like these :D

File: pages/data-entry-entity/measure-edit/MeasureEdit.js (hunk 2 above)
Comment (mateen-hussain) [reply pages/data-entry-entity/measure-edit/MeasureEdit.js:336]: Yea this was discussed with @leighjohnson-friday and changed accordingly.

File: pages/data-entry-entity/measure-edit/MeasureEdit.js [hunk 4]
```diff
@@ -125,29 +125,24 @@ class MeasureEdit extends Page {
...
-
-    const measureEntitiesMapped = this.mapMeasureFieldsToEntity(entities, themeCategory);
+    entities = await filterMetricsHelper.filterMetrics(this.req.user,entities); 
+    const measureEntitiesMapped = this.mapMeasureFieldsToEntity(entities, themeCategory); 
 
     // In certain case when a measure is the only item in the group, we need to up allow users to update the
     // overall value which is stored in the RAYG row.
-    return measureEntitiesMapped.reduce((data, entity) => {
+    const groupedEntitnes =  measureEntitiesMapped.reduce((data, entity) => {
       if(entity.filter === 'RAYG') {
```
Comment (leighjohnson-friday) [inline pages/data-entry-entity/measure-edit/MeasureEdit.js:134]: Minor Typo

**Review by leighjohnson-friday — APPROVED** [review]

File: pages/data-entry-entity/measure-edit/MeasureEdit.js (hunk 4 above)
Comment (mateen-hussain) [reply pages/data-entry-entity/measure-edit/MeasureEdit.js:134]: will fix this in next PR.

**Pull request merged**